packages/core/src/queries/application.ts

```diff
@@ -138,6 +138,19 @@ export const createApplicationQueries = (pool: CommonQueryMethods) => {
       and ${fields.type} = ${ApplicationType.Protected}
     `);
 
+  /**
+   * Find an protected application by its custom domain.
+   * the domain is stored in the `customDomains` field of the `protectedAppMetadata` field.
+   */
+  const findApplicationByProtectedAppCustomDomain = async (domain: string) =>
+    pool.maybeOne<Application>(sql`
+      select ${sql.join(Object.values(fields), sql`, `)}
+      from ${table}
+      where ${fields.protectedAppMetadata} ? 'customDomains'
+      and ${fields.protectedAppMetadata}->'customDomains' @> ${sql.jsonb([domain])}
+      and ${fields.type} = ${ApplicationType.Protected}
```

Comment:
do we have index for this jsonb field

Reply:
The field `customDomains` is optional, not sure if we can add index directly, maybe add the index later? Created LOG-8072 on this project to track.